monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,133 @@
+            operator_location = OperatorLocation(lat = operator_details['location']['latitude'], lng = operator_details['location']['longitude'])
+            operator_id = str(uuid.uuid4())
+
+            rid_flight_details = RIDFlightDetails(operator_id = operator_id, operator_location = operator_location, operation_description = flight_details['operation_description'] , serial_number = flight_details['serial_number'], registration_number = flight_details['registration_number'])
+
+            test_flight_details = TestFlightDetails(effective_after= effective_after,details = rid_flight_details)
+            test_flight = TestFlight(injection_id = str(uuid.uuid4()), telemetry= rid_state_data['flight_telemetry'], details_responses = test_flight_details)            
+            test_flight_deserialized = self.make_json_compatible(test_flight)
+            requested_flights.append(test_flight_deserialized)
+            test_payload = {'test_id': str(uuid.uuid4()), "requested_flights": requested_flights}        
+
+            all_test_payloads.append({'injection_url':uss['injection_url'], 'injection_payload': test_payload, 'injection_start_time_from_now_secs':uss['start_time_from_now_secs']})        
+        
+        return all_test_payloads
+
+
+class TestHarness():
+    ''' A class to submit Aircraft RID State to the USS test endpoint '''
+
+    def __init__(self, auth_spec:str, auth_url:str):
+        self.auth_spec = auth_spec
+        self.auth_url= auth_url
+        
+    def get_dss_session(self, auth_url:str, auth_spec:str):
+        ''' This method gets a DSS session using the monitoring tools that are provided in the DSS monitoring repository'''
+
+        auth_adapter = make_auth_adapter(auth_spec)
+        s = DSSTestSession(auth_url, auth_adapter)
+    
+        return s
+
+    async def submit_test(self,dss_session, injection_url,  test_payload):
+        print(f"Started: {time.strftime('%X')}")
+        print("Waiting %f seconds" % test_payload['injection_start_time_from_now_secs'])
+        await asyncio.sleep(test_payload['injection_start_time_from_now_secs'])            
+        response = dss_session.put(injection_url, data=test_payload['injection_payload'])
+        print(f"Ended: {time.strftime('%X')}")
+
+        if response.status_code == 200:
+            print("New test with ID %s created" % test_payload['injection_payload']['test_id'])
+        elif response.status_code ==409:
+            print("Test with ID %s already exists" % test_payload['injection_payload']['test_id'])
+        else: 
+            print(response.json())
+
+
+    
+    async def submit_payload_async(self, test_payloads):        
+        ''' This method submits the payload to the injection url by creating a DSSTestSession and then using that session to send the payload '''
+        for test_payload in test_payloads:
+            injection_url = test_payload['injection_url']        
+            auth_sub = urlparse(injection_url).netloc
```

Comment:
This should be unnecessary -- the `auth_spec` is specifying how the person running `rid_qualifier` gets their tokens, not how everyone else gets their tokens.  If USS1 is running `rid_qualifier`, tokens to access other USSs' systems will still indicate a `sub`scriber of "USS1" regardless of the audience the token is being sent to.

There should only be a single auth spec for a given invocation of rid_qualifier.  And, for efficiency, there should only be a single auth adapter instance for a given invocation of rid_qualifier.

Reply:
OK this is cleaned up now, thanks!